pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -218,6 +219,17 @@ func (s *Service) DeregisterInstanceFromAPIServerELB(i *infrav1.Instance) error
 	}
 
 	_, err = s.scope.ELB.DeregisterInstancesFromLoadBalancer(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case elb.ErrCodeAccessPointNotFoundException, elb.ErrCodeInvalidEndPointException:
+				// Ignoring LoadBalancerNotFound and InvalidInstance when deregistering
+				return nil
+			default:
+				return err
+			}
+		}
+	}
 	return err
```

Comment:
Since we are now handling the error above, this should probably explicitly return nil here now.
```suggestion
	return nil
```

Reply:
Totally agree - `err` will always be `nil` in this case, so it's easier on the reader to just return `nil`.